plugins/modules/win_ntp.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: 2020, Sakar Mehra (@sakar97) <sakar.mehra@gslab.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r'''
+---
+module: win_ntp
+short_description: Manage Windows NTP Server
+author: Sakar Mehra (@sakar97)
+requirements:
+  - This module requires Windows Server 2012R2 or Newer
+description:
+  - Manage Windows NTP Client
+  - Adds, Removes and Modifies NTP Server in windows machine.
+options:
+  peerlist:
+    description:
+      - Fully qualified domain name or IP address from which computer obtain timestamp.
+    type: list
+  type:
+    description:
+      - Indicates which peers to accept synchronization from.
+      - The default value on stand-alone clients and servers is NTP.
+    type: str
+    choices: [ NTP, NoSync, NT5DS, AllSync ]
+  enabled:
+    description:
+      - Indicates if the NtpClient provider is enabled in the current Time Service.
+      - The default value on stand-alone clients and servers is true.
+    type: bool
+    default: true
+  time_zone:
+    description:
+      - Set the time zone for the ntp server.
+    type: str
+    default: UTC
+  factory_default:
+    description:
+      - Ntp configuration are set to default with this option.
+    type: bool
+    default: false
+  cross_site_sync_flags:
+    description:
+      - Determines whether the service chooses synchronization partners outside the domain of the computer.
+      - Thsi value is ignored if l(type=NT5DS) is not set.
+    type: int
+    choices: [ 0, 1, 2 ]
+  sync_clock:
+    description:
+      - It resynchronize its clock as soon as possible.
+      - To update the ntp server with new peerlist.
+    elements: str
+    type: list
+    default: time.windows.com,0x9
```

Comment:
Can you explain this format any more, what's the `0x9` representing?

Reply:
Value “0x9” means – it use special interval + client mode association to the external time source. There are 0x1,0x2,0x4 and 0x8 as SpecialInterval,UseAsFallbackOnly,SymmetricActive,Client Mode flag respectively.

Reply:
So what we should have is to make this option `type: list` with `elements: dict`. Then those suboptions would be something like `source` and `flags` (or whatever relevant name it is`. Having end users deal with raw integers or bit flags is not ideal. Someone who is reading the tasks should have a generic idea what each parameter means without having to look at the docs. The module can easily parse those string values back to the relevant integer to deal with the getting and setting aspect internally.